### DISCUSSION TITLE
Is it possible to set maxAge for a particular query key so that only that query gets busted after maxAge?

### DISCUSSION THREAD

**Discussion by lisherwin** [TanStack/query#5389] [Q&A] [answered]
Hi React Query folks!

I am persisting my app's queries with `PersistQueryClientProvider` and I was wondering if there is a canonical way to tell React Query to bust data for a particular `queryKey` after some specified time (similar to `cacheTime` but for persisted queries). I know there is a `maxAge` `persistOption` that is used to tell React Query the maximum time it should persist the query cache, but that applies to the whole query cache. I am currently using a `maxAge` of `Infinity`, and I want to make sure that certain queries are not stored indefinitely so that we're only storing what we need on the client.

Sorry, no real code sandbox since this is more of an API question. Please let me know if I'm misunderstanding this. Thanks in advance!

**Comment by TkDodo** [comment]
no I don't think its possible currently

**Reply by cvincentcoleman** [reply]
Coulden't they set the `cacheTime` on the query and keep Infinity in the default options?

**Comment by TkDodo** [comment, marked as answer]
The new persister option on useQuery can do it: https://tanstack.com/query/v5/docs/react/plugins/createPersister

**Comment by hussam-001** [comment]
### Setting Custom `maxAge` for Each Query

Here’s an example of how to configure `PersistQueryClientProvider` to allow setting a custom `maxAge` for individual queries:

```jsx
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      gcTime: DEFAULT_MAX_AGE ,
      meta: { maxAge: DEFAULT_MAX_AGE }, // Default maxAge for all queries
    },
  },
});
```

```jsx
<PersistQueryClientProvider
    client={queryClient}
    persistOptions={{
      persister,
      maxAge: Infinity,
      dehydrateOptions: {
        shouldDehydrateMutation: () => false,
        shouldDehydrateQuery: (q) => {
          const { maxAge } = q.meta ?? {};
          if (typeof maxAge === "number") {
            return Date.now() - q.state.dataUpdatedAt < maxAge;
          }
          return false;
        },
      },
    }}
>
```

```jsx
useQuery({
  meta: { maxAge: CUSTOM_MAX_AGE },
  queryKey: ...,
  queryFn: ...
});
```


Note: persistOptions.maxAge must always be greater than or equal to CUSTOM_MAX_AGE. To ensure this, I set persistOptions.maxAge to Infinity. This prevents global expiration from interfering with query-specific configurations.

To manage default behavior, I configured the QueryClient to include a default meta value for queries
